x-pack/plugins/alerting/public/pages/maintenance_windows/helpers/convert_to_rrule.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Moment } from 'moment';
+import { RRule, RRuleFrequencyMap } from '../types';
+import { Frequency, ISO_WEEKDAYS_TO_RRULE } from '../constants';
+import { getNthByWeekday } from './get_nth_by_weekday';
+import { RecurringScheduleFormProps } from '../components/schema';
+import { getPresets } from './get_presets';
+
+export const convertToRRule = (
+  startDate: Moment,
+  timezone: string,
+  form?: RecurringScheduleFormProps
+): RRule => {
+  const presets = getPresets(startDate);
+
+  const rRule: RRule = {
+    dtstart: startDate.toISOString(),
+    tzid: timezone,
+  };
+
+  if (form) {
+    if (form.frequency !== 'CUSTOM') {
```

Comment:
I'm curious to know what you think, but I'm not a huge fan of having this union type with `'CUSTOM'` baked into the `Frequency` type because it's not really a frequency. I get we need this flag to indicate a custom RRule, but to me, that is only relevant as a part of the UI state. 

Maybe it's better to have a `frequencySelection` (or whatever name makes sense) field and collapse `frequency` with `customFrequency` into 1 field since both cases need to define a frequency, you might have some data duplication between `frequencySelection` and `frequency` but at least you can always rely on the `frequency` field being defined.

Reply:
I understand what you are saying, but I think with form lib this is not as easy to do and I would prefer to leave it the way it's written. Maybe if it's something we want to change later on we can!